src/render/View3D.cpp

```diff
@@ -498,6 +514,42 @@ void View3D::paintGL()
     glCheckError();
 }
 
+
+QImage View3D::renderScreenShot(int width, int height)
+{
+    if (m_badOpenGL)
+        return QImage();
+
+    double dPR = getDevicePixelRatio();
+    if (width < 0)
+        width = this->width();
+
+    if (height < 0)
+        height = this->height();
+
+    // Note: the reason for the following two lines and the subsequent downsampling of the
+    //       resulting image is to make the radius of rendered points consistent between the
+    //       gui and screenshot on displas with dPR != 1.
+    width *= dPR;
+    height *= dPR;
+
+    QOpenGLFramebufferObject framebuffer(QSize(width, height), QOpenGLFramebufferObject::Depth);
+    framebuffer.bind();
+    resizeViewport(width, height);
```

Comment:
This ends up calling `m_camera.setViewport()` which is wired into `restartRender()` via the `projectionChanged()` signal, so will probably force an unnecessary GUI redraw.  Will need some disentangling - perhaps just copying the required `glViewport()` stuff in here?  Really it's not a resize, so calling resize is a bit confusing anyway.